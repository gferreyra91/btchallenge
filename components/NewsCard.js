import React from 'react';
import {StyleSheet} from 'react-native';
import {Card, Button} from 'react-native-elements';

const NewsCard = ({news, onPressNext, onPressRemove}) => {
  return (
    <Card>
      <Card.Title>{news.title}</Card.Title>
      <Card.Divider />
      <Card.Image source={{uri: news.urlToImage}} />
      <Card.Divider />
      <Button
        title="View"
        onPress={() => onPressNext()}
        buttonStyle={styles.nextButton}
      />
      <Card.Divider />
      {onPressRemove && (
        <Button
          title="Remove"
          onPress={() => onPressRemove()}
          buttonStyle={styles.removeButton}
        />
      )}
    </Card>
  );
};

export default NewsCard;

const styles = StyleSheet.create({
  removeButton: {
    backgroundColor: 'red',
  },
});
