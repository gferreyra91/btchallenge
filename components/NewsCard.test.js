import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import NewsCard from './NewsCard';
import {MOCK_ARTICLES} from '../mocks';
import {fireEvent, render} from '@testing-library/react-native';

it('renders correctly', () => {
  const mockOnPressNext = jest.fn();
  const mockOnPressRemove = jest.fn();
  const {getByTestId, toJSON} = render(
    <NewsCard
      news={MOCK_ARTICLES[0]}
      onPressNext={mockOnPressNext}
      onPressremove={mockOnPressRemove}
    />,
  );
  const cardTitle = getByTestId('cardTitle');
  expect(cardTitle.children[0]).toBe(MOCK_ARTICLES[0].title);
  const cardImage = getByTestId('RNE__Image');
  expect(cardImage.props.source.uri).toBe(MOCK_ARTICLES[0].urlToImage);
  expect(toJSON()).toMatchSnapshot();
});
