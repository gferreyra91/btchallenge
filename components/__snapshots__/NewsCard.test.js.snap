// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#43484d",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
      testID="cardTitle"
    >
      Third stimulus check updates: Biden, Yellen pan GOP virus aid as too small - Chicago Tribune
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#bcbbc1",
          "height": 0.5,
          "marginBottom": 15,
        }
      }
    />
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Object {
            "uri": "https://www.chicagotribune.com/resizer/UCoTDELQpB6AOxfuw4TLxFyP--c=/1200x0/top/cloudfront-us-east-1.images.arcpublishing.com/tronc/4HD5RMFG6OGSRWJEZ73XNN22ME.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 150,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": null,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
        transitionDuration={360}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 150,
              "justifyContent": "center",
              "width": null,
            }
          }
          testID="RNE__Image__placeholder"
        />
      </View>
      <View
        style={
          Object {
            "height": 150,
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#bcbbc1",
          "height": 0.5,
          "marginBottom": 15,
        }
      }
    />
    <View
      style={
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            View
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#bcbbc1",
          "height": 0.5,
          "marginBottom": 15,
        }
      }
    />
  </View>
</View>
`;
